.github/workflows/tests.yml

```diff
@@ -247,141 +327,159 @@ jobs:
-          # when locally dowloading and comparing results of different workflow runs.
-          name: test-results-${{ github.sha }}-linux-${{ matrix.platform }}-qemu-${{ matrix.default-channel }}-${{ matrix.python-version }}
-          path: |
-            .coverage
-            test-report.xml
-          retention-days: 1
+          ${{ matrix.image }}
+          bash -lc
+          ". /opt/conda/etc/profile.d/conda.sh
+          && conda create
+            --name test
+            --yes
+            --file tests/requirements.txt
+            --file tests/requirements-${{ runner.os }}.txt
+            --file tests/requirements-ci.txt
+            python=${{ matrix.python-version }}
+          && conda activate test
+          && conda info --verbose
+          && conda config --show-sources
+          && conda list --show-channel-urls
+          && python -m pytest tests/test_api.py::test_DepsModifier_contract"
 
   # macos test suite
   macos:
     # only run test suite if there are code changes
     needs: changes
     if: needs.changes.outputs.code == 'true'
 
-    runs-on: ${{ (matrix.arch == 'osx-64' && 'macos-latest') || 'macos-14' }}
+    runs-on: ${{ (matrix.arch == 'osx-64' && 'macos-13') || 'macos-14' }}
+    defaults:
+      run:
+        # https://github.com/conda-incubator/setup-miniconda#use-a-default-shell
+        shell: bash -el {0}  # bash exit immediately on error + login shell
     strategy:
       fail-fast: false
       matrix:
+        # test lower version (w/ osx-64 & defaults & unit tests) and upper version (w/ osx-arm64 & conda-forge & integration tests)
         arch: [osx-64, osx-arm64]
-        default-channel: ['defaults', 'conda-forge']
-        python-version: ['3.10']
-        test-type: ['unit', 'integration']
-        test-group: ['1', '2', '3']
-        # fewer test splits for quicker unit tests, keep 3 splits for integration
-        test-splits: [2, 3]
+        python-version: ['3.8', '3.12']
+        default-channel: [defaults, conda-forge]
+        test-type: [unit, integration]
+        test-group: [1, 2, 3]
         exclude:
+          - arch: osx-64
+            python-version: '3.12'
+          - arch: osx-64
+            default-channel: conda-forge
+          - arch: osx-64
+            test-type: integration
+          - arch: osx-arm64
+            python-version: '3.8'
+          - arch: osx-arm64
+            default-channel: defaults
+          - arch: osx-arm64
+            test-type: unit
           - test-type: unit
-            test-group: '3'
-          - test-type: unit
-            test-splits: 3
-          - test-type: integration
-            test-splits: 2
-          # odd choice kept for parity with previous commit
-          - test-type: integration
-            default-channel: 'conda-forge'
-          - test-type: unit
-            default-channel: 'defaults'
+            test-group: 3
```

Comment:
Is this line continuing to do fewer unit test groups on OSX to preserve limited resources?

Reply:
yup, the macOS matrix only runs 5 jobs:

- osx-64 + Python 3.8 + defaults + unit 1
- osx-64 + Python 3.8 + defaults + unit 2
- osx-arm64 + Python 3.12 + conda-forge + integration 1
- osx-arm64 + Python 3.12 + conda-forge + integration 2
- osx-arm64 + Python 3.12 + conda-forge + integration 3